src/Psy/Configuration.php

```diff
@@ -1162,15 +1167,30 @@ public function setUpdateCheck($interval)
     /**
      * Get a cache file path for the update checker.
      *
-     * @return string
+     * @return string|false Return false if config file/directory is not writable
      */
     public function getUpdateCheckCacheFile()
     {
         $dir = $this->configDir ?: ConfigPaths::getCurrentConfigDir();
+
+        if (!is_writable($dir)) {
```

Comment:
this problem needs to be solved for the history file as well. any thoughts on a more general solution?